File: .yalc/@entur/atb-mobile-client-sdk/token/token-core-android-lib/src/main/java/no/entur/abt/android/token/keystore/TokenKeyStore.java
package no.entur.abt.android.token.keystore;

import java.security.KeyStoreException;
import java.security.NoSuchAlgorithmException;
import java.security.PrivateKey;
import java.security.UnrecoverableEntryException;
import java.security.cert.Certificate;

import no.entur.abt.android.token.TokenContext;

public interface TokenKeyStore<T> {

	TokenTrustChain createEncryptionKey(TokenContext<T> tokenContext, String tokenId, byte[] attestationChallenge) throws TokenKeystoreException;

	TokenTrustChain getEncryptionTrustChain(TokenContext<T> tokenContext, String tokenId)
			throws TokenKeystoreException, UnrecoverableEntryException, NoSuchAlgorithmException, KeyStoreException;

	Certificate getEncryptionCertificate(TokenContext<T> tokenContext, String tokenId) throws KeyStoreException, TokenKeystoreException;

	PrivateKey getEncryptionPrivateKey(TokenContext<T> tokenContext, String tokenId)
			throws UnrecoverableEntryException, NoSuchAlgorithmException, KeyStoreException, TokenKeystoreException;

	TokenTrustChain createSignatureKey(TokenContext<T> tokenContext, String tokenId, byte[] attestationChallenge) throws TokenKeystoreException;

	TokenTrustChain getSignatureTrustChain(TokenContext<T> tokenContext, String tokenId)
			throws TokenKeystoreException, UnrecoverableEntryException, NoSuchAlgorithmException, KeyStoreException;

	void setSignatureCertificate(TokenContext<T> tokenContext, String tokenId, PrivateKey key, Certificate certificate)
			throws KeyStoreException, TokenKeystoreException;

	Certificate getSignatureCertificate(TokenContext<T> tokenContext, String tokenId) throws KeyStoreException, TokenKeystoreException;

	PrivateKey getSignaturePrivateKey(TokenContext<T> tokenContext, String tokenId)
			throws UnrecoverableEntryException, NoSuchAlgorithmException, KeyStoreException, TokenKeystoreException;

	void removeTokens() throws TokenKeystoreException;

	void removeTokens(TokenContext<T> tokenContext) throws TokenKeystoreException;

	boolean hasTokens() throws TokenKeystoreException;

	boolean hasTokens(TokenContext<T> tokenContext) throws TokenKeystoreException;

	void removeToken(TokenContext<T> tokenContext, String tokenId) throws TokenKeystoreException;

}

File: .yalc/@entur/atb-mobile-client-sdk/token/token-core-ios-lib/AbtTokenCoreSDK/keystore/TokenKeyStore.swift
import Foundation

public protocol TokenKeyStore {
    // MARK: Encryption

    func createEncryptionKey(tokenContext: TokenContext, tokenId: String) -> Result<KeyPair, TokenKeyStoreError>

    func getEncryptionPrivateKey(tokenContext: TokenContext, tokenId: String) -> PrivateKey?

    func getEncryptionKeyPair(tokenContext context: TokenContext, tokenId: String) -> KeyPair?

    func getEncryptionCertificate(tokenContext: TokenContext, tokenId: String) -> SecCertificate?

    // MARK: Signature

    func createSignatureKey(tokenContext: TokenContext, tokenId: String) -> Result<KeyPair, TokenKeyStoreError>

    @available(iOS 14.0, *)
    func createSignatureKey(tokenContext context: TokenContext, tokenId: String, onComplete callback: @escaping (Result<String, TokenKeyStoreError>) -> Void)

    @available(iOS 14.0, *)
    func getSignatureKey(tokenContext context: TokenContext, tokenId: String) -> String?

    func getSignaturePrivateKey(tokenContext: TokenContext, tokenId: String) -> PrivateKey?

    func getSignatureKeyPair(tokenContext context: TokenContext, tokenId: String) -> KeyPair?

    func getSignatureCertificate(tokenContext: TokenContext, tokenId: String) -> SecCertificate?

    func setSignatureCertificate(tokenContext: TokenContext, tokenId: String, certificate: Data) -> SecCertificate?

    // MARK: Tokens

    func removeTokens(tokenContext: TokenContext)

    func hasTokens() -> Bool

    func hasTokens(tokenContext: TokenContext) -> Bool

    func removeToken(tokenContext: TokenContext, tokenId: String)
}
